src/btc.rs

```diff
@@ -192,14 +228,202 @@ impl UnsignedTx {
+                    num_keys,
+                    threshold,
+                    &mut OsRng,
+                )
+            })
+            .collect();
+
+        let polys = match test_helpers::dkg(&mut signers, &mut OsRng) {
+            Ok(polys) => polys,
+            Err(secret_errors) => {
+                panic!("Got secret errors from DKG: {secret_errors:?}");
+            }
+        };
+
+        let mut signing_set = [signers[0].clone(), signers[1].clone(), signers[3].clone()].to_vec();
+        let key_ids = signing_set
+            .iter()
+            .flat_map(|s| s.get_key_ids())
+            .collect::<Vec<u32>>();
+        let mut sig_agg = v2::Aggregator::new(num_keys, threshold);
+        sig_agg.init(&polys).expect("aggregator init failed");
+        let wsts_public_key = sig_agg.poly[0];
+        let wsts_public_key_bytes = wsts_public_key.x().to_bytes();
+
+        // the sBTC accept script must be of the form
+        //
+        //     <sbtc_payload> DROP <signers_pubkey> CHECKSIGVERIFY
+        //
+        // with 160-bytes of <sbtc_payload>
+
+        let sbtc_payload = [255u8; 160];
+        let mut push_bytes = PushBytesBuf::new();
+        push_bytes
+            .push(sbtc_payload.len().try_into().unwrap())
+            .unwrap();
+        push_bytes.extend_from_slice(&sbtc_payload).unwrap();
+
+        let accept_script = Builder::new()
+            .push_slice(push_bytes)
+            .push_opcode(OP_DROP)
+            .push_slice(wsts_public_key_bytes)
+            .push_opcode(OP_CHECKSIG)
+            .into_script();
+
+        println!("Accept script {accept_script}");
+
+        // Generate keys for the depositor
+        let secp = Secp256k1::new();
+        let depositor_secret_key = secp256k1::SecretKey::new(&mut OsRng);
+        let depositor_keypair = secp256k1::Keypair::from_secret_key(&secp, &depositor_secret_key);
+        let (depositor_public_key, _) = depositor_keypair.x_only_public_key();
+
+        // the reject script must be of the form
+        //
+        //     <lock_time> CHECKSEQUENCEVERIFY <reclaim script>
+        //
+        let reject_script = Builder::new()
+            .push_lock_time(absolute::LockTime::ZERO)
+            .push_opcode(OP_CSV)
+            .push_opcode(OP_DROP)
+            .push_x_only_key(&depositor_public_key)
+            .push_opcode(OP_CHECKSIG)
+            .into_script();
+
+        println!("Reject script {reject_script}");
+
+        // BIP-341 defines the NUMS field element as the hash of the x-coord of G
+        let nums_x_data =
+            hex::decode("50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0")
+                .unwrap();
+        let nums_x = curve::field::Element::try_from(nums_x_data.as_slice()).unwrap();
+        let nums_key_point = curve::point::Point::lift_x(&nums_x).unwrap();
+        let nums_public_key =
+            bitcoin::XOnlyPublicKey::from_slice(&nums_key_point.x().to_bytes()).unwrap();
```

Comment:
nit: can't we just do this?
```suggestion
            bitcoin::XOnlyPublicKey::from_slice(&nums_x_data).unwrap();
```

Reply:
Good catch, `XOnlyPublicKey::from_slice` does the `lift_x` for us.

Reply:
Fixed